lib/toolkit.ts

```diff
@@ -1,14 +1,10 @@
 import Response from './response.ts';
 import * as bang from './bang.ts';
-import { ResponseBody } from './types.ts';
+import { ToolkitInterface, FileHandlerOptions, ResponseBody } from './types.ts';
 import isPathInside from './util/is-path-inside.ts';
 import { mime } from '../dependencies.ts';
 
-interface FileHandlerOptions {
-    confine: boolean | string
-}
-
-export default class Toolkit {
+export default class Toolkit implements ToolkitInterface {
```

Comment:
I'm not sure I understand what value this change provides. How exactly does this change the `deno doc` output?

Reply:
It adds an `interface` under `namespace types` that describes `Toolkit`.
```sh
const router
const server
class Router
class Server
namespace types
  ...
  interface ToolkitInterface
  ...
```

Reply:
I added it so that the docs could explain what will be available on the `Toolkit`.
If you think it isn't necessary I can remove it.

Reply:
I think it's great that it shows which helpers are available. The thing is, I thought `deno doc` was capable of describing the shape of a class without an interface. Maybe its not showing the methods because they lack TS Doc comments?

Also, is there a reason why you want to do it this way with `Toolkit`, but not `Router`, for example? Seems like whichever way we go on this, it should be consistent.

Reply:
- `deno doc` can describe the shape of a class, without an interface, so long as the class is exported from:
    - The `deno doc` entry point file (i.e. `main.ts`).
      -  Or whichever file is put in the deno doc badge on the readme.
  - I was putting interfaces in `types.ts` to:
    - Avoid re-exporting anything from `main.ts` that existed in `lib`
    - Avoid adding `jsdoc` to the class, and adding it to the `types` instead. (may not be what you prefer though)
    - The `jsdoc` could live on the `interface` as well as the `class`, but that may become burdensome. 
  - E.g. `jsdoc` entries will need to get added to the factories `server` and `router` inside of `main.ts`.

Reply:
Ah, okay, I understand. You made it an interface because I said I didn't want to expose classes that users shouldn't instantiate themselves. That makes sense.

I guess I didn't think this all the way through. Having a seperate interface is probably more trouble than it's worth, just for the sake of documentation. I take back what I said earlier - let's expose the class, same as `Router`, etc.